### PR TITLE
fix(AppMain): only show banner when disconnected

### LINKED ISSUE #7416: [Banners] Only show “Connected” banner if “Disconnected” happened.
### Description
So there should be no “Connected” banner at startup.

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes #7416

### What does the PR do
Banners: only show banner when disconnected

### Affected areas
Banners

### CODE REVIEW

**Review by caybro — COMMENTED** [review]
How does it auto hide when the user is back online?

**Review by caybro — CHANGES_REQUESTED** [review]
I think this will break: 
- the autohiding when the user comes back online
- it will still say "Disconnected" even when online

**Comment by alexandraB99** [comment]
> I think this will break:
> 
>     * the autohiding when the user comes back online
> 
>     * it will still say "Disconnected" even when online

you're right! added back the check, if it's connected to hide.

**Review by caybro — APPROVED** [review]

**Review by igor-sirotin — COMMENTED** [review]
I might be wrong, but I thought we agreed that:
the banner shouldn't show "Connected" at startup, BUT it should change to green "connected" when the connection is restored.

File: ui/app/mainui/AppMain.qml [hunk 1]
```diff
@@ -503,31 +505,22 @@ Item {
 
                     ModuleWarning {
                         readonly property bool isConnected: bannersLayout.isConnected
-
                         objectName: "connectionInfoBanner"
                         Layout.fillWidth: true
-                        text: isConnected ? qsTr("Connected") : qsTr("Disconnected")
```
Comment (micieslak) [inline ui/app/mainui/AppMain.qml:509]: If I understand correctly, "Connected" is also needed, when we go back online after being some time offline.

File: ui/app/mainui/AppMain.qml [hunk 2]
```diff
@@ -503,31 +505,22 @@ Item {
...
-                        }
                         onCloseClicked: {
                             hide();
                         }
                         onHideStarted: {
                             bannersLayout.connectedBanner = null
                         }
+                        onIsConnectedChanged: {
+                            if (isConnected) {
+                                hide();
```
Comment (micieslak) [inline ui/app/mainui/AppMain.qml:521]: It seems to be never shown

**Review by micieslak — CHANGES_REQUESTED** [review]
Some questions/doubts

**Review by igor-sirotin — CHANGES_REQUESTED** [review]
We agreed with @alexandraB99 that there should still be "Connected" banner, we just don't want to see it at application startup 🙂

**Comment by alexandraB99** [comment]
> We agreed with @alexandraB99 that there should still be "Connected" banner, we just don't want to see it at application startup 🙂

@igor-sirotin changed it as agreed, however I am not sure this is the correct variable to set to true as I can still see it on startup but I will check this further. Marking as `do not merge` for now

**Comment by elina2015** [comment]
@alexandraB99 I still can see the banner 'Connected' on the startup. After disconnecting and connecting again, the banner doesn't appear. Could you kindly check?

**Review by elina2015 — CHANGES_REQUESTED** [review]
Need fixes

**Comment by alexandraB99** [comment]
@elina2015, @micieslak "After disconnecting and connecting again, the banner doesn't appear." -> this is not a result of this PR, it also happens in master. @igor-sirotin setting `isConnected` to true in nim didn't have any affect as it seems that local `isConnected` property was false until Component initialization. this solution seems working fine.

**Review by elina2015 — APPROVED** [review]
Tested and approved

**Review by micieslak — CHANGES_REQUESTED** [review]

File: ui/app/mainui/AppMain.qml [hunk 3]
```diff
@@ -580,7 +580,7 @@ Item {
                         }
 
                         Component.onCompleted: {
-                            updateState()
+                            connectedBannerComponent.isConnected = bannersLayout.isConnected;
```
[the remark this answers is not in the document]
Comment (alexandraB99) [reply ui/app/mainui/AppMain.qml]: good catch! I missed the reference to a Component. Code updated, please have a second look!

[the remark this answers is not in the document]
Comment (micieslak) [reply ui/app/mainui/AppMain.qml]: Thank you @alexandraB99 for amending! Now it looks and works nice :+1:

**Review by micieslak — APPROVED** [review]

**Review by igor-sirotin — APPROVED** [review]
Looks good, thanks!

**Pull request merged**